### PR TITLE
Adds support for showing warning icon in StatefulSet

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen Singh <harkishensingh@hotmail.com>

Fixes: #4205 

**Explanatory Image:**

Screenshot from 2020-04-09 12-41-25

### CODE REVIEW

**Comment by floreks** [comment]
I'd argue that when desired nr of pods is different than actual number it does not necessarily mean there is something wrong with it. It might be upscaling/downscaling. This case is definitely not a warning.

**Comment by Harkishen-Singh** [comment]
@floreks sure. I actually implemented that way since I did not any other notable things in response from the API (`getData()`). So, I thought that the warning symbol should be kind of at that can signify anything different than the desired value (noticed in gke, I think so). 

What do you suggest should be the recommended way here? @floreks

**Comment by floreks** [comment]
Our error state is actually a warning state. If there are warning/error events associated with any resource, an error icon will appear and proper message will be available. I don't think a warning status is needed here.

**Comment by Harkishen-Singh** [comment]
@floreks does that mean, the _green check_ in `0/1` Pods is acceptable?

**Comment by floreks** [comment]
This is weird because 0/1 should mean that there is a pending pod waiting to be created. You can extend the pending/success checks to check for pending pods and if desired/actual pods match. In such case it should show pending status.

**Comment by Harkishen-Singh** [comment]
Got your point. However, in this case, the pods will forever remain in a pending case. Hence I thought that showing a warning or alert icon will be required, as the issues say.

**Comment by Harkishen-Singh** [comment]
@floreks updated the logic as per discussion.

Screenshot from 2020-04-09 12-41-25

File: src/app/frontend/common/components/resourcelist/statefulset/component.ts [hunk 1]
```diff
@@ -74,24 +73,30 @@ export class StatefulSetListComponent extends ResourceListWithStatuses<
...
-    return resource.podInfo.warnings.length === 0 && resource.podInfo.pending > 0;
+    return (resource.podInfo.warnings.length === 0 && resource.podInfo.pending > 0) || (resource.podInfo.running !== resource.podInfo.desired);
   }
 
   isInSuccessState(resource: StatefulSet): boolean {
-    return resource.podInfo.warnings.length === 0 && resource.podInfo.pending === 0;
+    return resource.podInfo.warnings.length === 0 && resource.podInfo.pending === 0 && (resource.podInfo.running === resource.podInfo.desired);
+  }
+
+  getDisplayStatus(podInfo: PodInfo): string {
```
Comment (floreks) [inline src/app/frontend/common/components/resourcelist/statefulset/component.ts:88]: Remove that. Status column is reserved only for pods.

File: src/app/frontend/common/components/resourcelist/statefulset/component.ts [hunk 2]
```diff
@@ -74,24 +73,30 @@ export class StatefulSetListComponent extends ResourceListWithStatuses<
...
-    return resource.podInfo.running !== resource.podInfo.desired;
-  }
-
   isInErrorState(resource: StatefulSet): boolean {
     return resource.podInfo.warnings.length > 0;
   }
 
   isInPendingState(resource: StatefulSet): boolean {
-    return resource.podInfo.warnings.length === 0 && resource.podInfo.pending > 0;
+    return (resource.podInfo.warnings.length === 0 && resource.podInfo.pending > 0) || (resource.podInfo.running !== resource.podInfo.desired);
```
Comment (floreks) [inline src/app/frontend/common/components/resourcelist/statefulset/component.ts:81]: Pending and desired/running part should be taken into common parenthesis. It should be

_if there are no warnings and (there are pending pods or running !== desired)_

Adjust success state check accordingly.

File: src/app/frontend/common/components/workloadstatus/component.ts [hunk 3]
```diff
@@ -35,6 +35,11 @@ export class WorkloadStatusComponent {
   @Input() resourcesRatio = emptyResourcesRatio;
   colors: string[] = [];
 
+  constructor() {
```
Comment (floreks) [inline src/app/frontend/common/components/workloadstatus/component.ts:38]: Remove that.

**Comment by Harkishen-Singh** [comment]
@floreks done.

**Pull request merged**